searx/version.py

```diff
@@ -59,25 +58,9 @@ def get_git_url_and_branch():
 
 
 def get_git_version():
-    try:
-        tag = subprocess_run("git describe HEAD")
-        # a. HEAD is on tag name, example: tag = "v1.0.1"
-        # b. HEAD is not a tag name, example "<tag>-<distance>-g<commit>"
-        tag_version, tag_distance, tag_commit = (tag.split("-") + ["", ""])[:3]
-        if re.match(r"v[0-9]+\.[0-9]+\.[0-9]+", tag_version):
-            # tag_version "v1.0.0" becomes "1.0.0" (without the v)
-            # other patterns are kept untouched
-            tag_version = tag_version[1:]
-        # remove "g" prefix from tag_commit
-        if tag_commit and tag_commit[0] == "g":
-            tag_commit = tag_commit[1:]
-        # set git_version to "1.0.0-590-0686e274" or '1.0.0'
-        git_version = "-".join(filter(bool, [tag_version, tag_distance, tag_commit]))
-    except subprocess.CalledProcessError:
-        # fall back to "YYYY.MM.DD.Hash" if there is no tag at all
-        git_version = subprocess_run(r"git show -s --format='%as-%h'")
-        # PEP 440: replace - with .
-        tag_version = git_version = git_version.replace("-", ".")
+    git_commit_date = subprocess_run(r"git show -s --format='%as'").replace("-", ".")
+    git_commit_hash = subprocess_run(r"git show -s --format='-%h'")
```

Comment:
Instead of the Auth-Date [%as](https://git-scm.com/docs/git-show#Documentation/git-show.txt-emasem) we should use the Commit-Date [%cs](https://git-scm.com/docs/git-show#Documentation/git-show.txt-emcsem).  

When a fork is rebased, the Author-Date is unchanged while the Commit-Date is updated. By example my production branch [darmarit.org](https://github.com/return42/searxng/commits/darmarit.org) has a [ [brand] commit ](https://github.com/return42/searxng/commit/0ce7fa3f161b083c6716aaeda3a65e4fed555993) on top.  When I update this branch, I rebase this branch on [SearXNG's master](https://github.com/searxng/searxng/tree/master). 

```
$ git show -s --format='%as'
2021-10-27

$ git show -s --format='%cs'
2022-02-21
```

@dalf: what do you think about `-` versus `.` in the date (year)? .. I prefer the `-` from the git command:

```
$ git show -s --format='%cs-%h'
2022-02-21-0ce7fa3f
```

.. the dot-nation can be more easily confused with the local date notation of the countries I think.

Reply:
After reading https://stackoverflow.com/questions/11856983/why-git-authordate-is-different-from-commitdate I like author date more;

If a user has a fork and they add commits from SearXNG upstream the version string is going to update to a newer version when using commit date, even tho there are no new features in the commit compared to the older version string in SearXNG upstream. With the author date we make sure that the version date stays the same even in forks for a commit...

Also I would propose as a version string format: `YYYY-MM-DD_hash`...

Reply:
> @dalf: what do you think about - versus . in the date (year)? .. I prefer the - from the git command:

The dot notation is used by
* Arch: https://archlinux.org/releng/releases/
* Invidious: https://api.invidious.io/
* I can't find a project using the dash notation.

I guess the dot notation is expected for a version.

So, I would choose the dots rather than the dash.

---

lines 61 and 62 can be written in one line:

```python
tag_version = subprocess_run(r"git show -s --format='%as-%h'").replace("-", ".", 2)
```
(or with `replace("-", ".", 2)` is the dash notation is chosen).

Reply:
> convinced me that commit date is better +1

and @dalf convinced me that dot notation for year notation is better :smiley: 

@tiekoetter could you update the PR?

Reply:
Changes:
- Change author date to commit date
- Keep YYYY.MM.DD-commit_hash
- Rewrite line 61 and 62 into one line.

Is this correct?